docs/README-Extending.md

```diff
@@ -0,0 +1,136 @@
+
+## VSCode Extender Package
+
+Extender packages are created as new VSCode extensions. Some useful links below:
+
+- [Using extensions in Visual Studio Code](https://code.visualstudio.com/docs/introvideos/extend) 
+- [VSCode Extension API](https://code.visualstudio.com/api) 
+- [vscode-extension-samples](https://github.com/Microsoft/vscode-extension-samples) 
+
+When creating an extension for Zowe Explorer it is necessary to specify that dependency in the package.json file as follows:
+```json	
+    "extensionDependencies": [
+		"zowe.vscode-extension-for-zowe"
+	],
+
+```
+The addition of this script also enforces that the Zowe Explorer is a pre-requisite package on the users machine which will install it if necessary 
+when the extender package is installed. This also ties the activate functionality of both packages enabling direct access to the Zowe Explorer API
+
+## Contextual Hooks
+
+To enable menus, extenders can reference the context values associated with the individual Tree Items. Zowe Extender uses a strategy of adding and removing 
+elements of context to the individual context values if that imparts additional information that could assist with menu triggers. In the example below we
+are referencing a Job type tree item that has additional information indicated by the *_rc* context. This can be used by an extender to trigger a specifc menu.   
+
+```json
+		"menus": {
+			"view/item/context": [
+				{
+					"when": "view == zowe.jobs && viewItem =~ /^job.*/ && viewItem =~ /^.*_rc=CC.*/",
+					"command": "zowe.testmule.retcode",
+					"group": "4_workspace"
+				}
+            ],
+```
+Notice the syntax we use is for the context value or viewItem above is a regular expression as denoted by the "=~" equal test. Using regular expressions 
+to describe context allows more meaning to be embedded in the context.
+
+Additional contexts are being indicated by the use of a preceding underscore character. These need to be added in the Zowe Explorer so the process to get them added 
+is by raising a [git issue](https://github.com/zowe/vscode-extension-for-zowe/issues) and requesting a specific hook.
+
+
+## Zowe Explorer Extender API
+
+The following code snippet shows how to gain access to the published Zowe Explorer extender API.
+
+```javascript
+export function activate(context: vscode.ExtensionContext) {
+    const EXTENDER_API;
+    const baseExt = vscode.extensions.getExtension('zowe.vscode-extension-for-zowe');
+    try {
+        if (baseExt && baseExt.exports) {
+            EXTENDER_API = baseExt.exports.getExplorerExtenderApi();
+            return;
+        }
+    } catch (error) {
+        vscode.window.showWarningMessage("Unable to access Zowe Explorer API");
+    }
+}
+```
+
+Details of function available in the Api can be found by looking in the [Zowe Explorer source](https://github.com/zowe/vscode-extension-for-zowe/blob/master/src/api/ZoweExplorerExtender.ts) 
```

Comment:
Details about the functions available in the API can be found by looking in the [Zowe Explorer source](https://github.com/zowe/vscode-extension-for-zowe/blob/master/src/api/ZoweExplorerExtender.ts). Initially, the following access methods were added to provide access to the appropriate profiles: